### PR TITLE
Fix bugs related to multiple slbs

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

1. If the vmSet is changed from non-primary to the primary vmSet, we need to make sure the dedicated LB is deleted.
2. In some cases, the private IP of an internal service may change when switching from one LB to another. We need to re-use the previous private IP.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Bug 2 cannot be reproduced, I have just hit it once.

#### Does this PR introduce a user-facing change?

```release-note
fix: remove agent pool LB if that vmSet is changed to primary vmSet
fix: reuse previous private IP address when changing load balancers
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -262,6 +262,10 @@ func (az *Cloud) shouldChangeLoadBalancer(service *v1.Service, currLBName, clust
 	// if the current LB is what we want, keep it
 	lbName := strings.TrimSuffix(currLBName, consts.InternalLoadBalancerNameSuffix)
 	if strings.EqualFold(lbName, vmSetName) {
+		if strings.EqualFold(az.VMSet.GetPrimaryVMSetName(), vmSetName) {
```
Comment (feiskyer) [inline pkg/provider/azure_loadbalancer.go:265]: should we check clusterName here as well? If clusterName==vmSetName, then no need to change the slb.

Comment (feiskyer) [reply pkg/provider/azure_loadbalancer.go:265]: it's possible on unmanaged clusters.

Comment (nilo19) [reply pkg/provider/azure_loadbalancer.go]: updated

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**